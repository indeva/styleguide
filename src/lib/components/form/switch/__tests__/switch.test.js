import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import SwitchControlled from 'test-helpers/controlleds/switch';
import Switch from '..';

describe('Switch', () => {
  const defaultProps = {
    name: 'name',
    value: 'value',
  };

  describe('snapshot tests', () => {
    describe('when pass only required props', () => {
      it('renders switch properly', () => {
        const component = renderer.create(<Switch {...defaultProps} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });

    describe('when has label property', () => {
      it('renders switch with label', () => {
        const component = renderer.create(<Switch {...defaultProps} label="label" />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });

    describe('when has checked property', () => {
      it('renders switch with checked true', () => {
        const component = renderer.create(<Switch {...defaultProps} checked />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });

    describe('when has disabled property', () => {
      it('renders switch with disabled true', () => {
        const component = renderer.create(<Switch {...defaultProps} disabled />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('behavior tests', () => {
    describe('when click on unchecked switch', () => {
      it('the input should be checked', () => {
        const wrapper = mount(<SwitchControlled {...defaultProps} />);

        wrapper
          .find('.switch__input')
          .simulate('change', { target: { checked: true } });

        expect(wrapper.find('.switch__input').props().checked).toBeTruthy();
      });
    });

    describe('when click on checked switch', () => {
      it('the input should be checked', () => {
        const wrapper = mount(<SwitchControlled {...defaultProps} checked />);

        wrapper
          .find('.switch__input')
          .simulate('change', { target: { checked: false } });

        expect(wrapper.find('.switch__input').props().checked).toBeFalsy();
      });
    });
  });
});
