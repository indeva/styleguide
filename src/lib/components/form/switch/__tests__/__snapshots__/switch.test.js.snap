// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch snapshot tests when has checked property renders switch with checked true 1`] = `
<div
  className="switch"
>
  <input
    checked={true}
    className="switch__input"
    disabled={null}
    id="name"
    name="name"
    onChange={null}
    type="checkbox"
    value="value"
  />
  <span
    className="switch__checked"
  />
</div>
`;

exports[`Switch snapshot tests when has disabled property renders switch with disabled true 1`] = `
<div
  className="switch"
>
  <input
    checked={null}
    className="switch__input"
    disabled={true}
    id="name"
    name="name"
    onChange={null}
    type="checkbox"
    value="value"
  />
  <span
    className="switch__checked"
  />
</div>
`;

exports[`Switch snapshot tests when has label property renders switch with label 1`] = `
<div
  className="switch"
>
  <input
    checked={null}
    className="switch__input"
    disabled={null}
    id="name"
    name="name"
    onChange={null}
    type="checkbox"
    value="value"
  />
  <span
    className="switch__checked"
  />
  <label
    className="switch__label"
    htmlFor="name"
  >
    label
  </label>
</div>
`;

exports[`Switch snapshot tests when pass only required props renders switch properly 1`] = `
<div
  className="switch"
>
  <input
    checked={null}
    className="switch__input"
    disabled={null}
    id="name"
    name="name"
    onChange={null}
    type="checkbox"
    value="value"
  />
  <span
    className="switch__checked"
  />
</div>
`;
